nautobot/core/templates/inc/nav_menu.html

```diff
@@ -32,30 +62,32 @@
                                             {% for item_link, item_details in group_details.items.items %}
                                                 {% if request.user|has_one_or_more_perms:item_details.permissions or not "HIDE_RESTRICTED_UI"|settings_or_config %}
                                                     <li {% if not request.user|has_perms:item_details.permissions %} class="disabled"{% endif %}>
-                                                        <div class="buttons pull-right">
-                                                            {% for button_title, button_details in item_details.buttons.items %}
-                                                                {% if request.user|has_perms:button_details.permissions %}
-                                                                    {% comment %}
-                                                                        Use 'url xxx as variable' so that an invalid
-                                                                        link doesn't throw a NoReverseMatch exception.
-                                                                    {% endcomment %}
-                                                                    {% url button_details.link as button_url %}
-                                                                    {% if button_url %}
-                                                                        <a href="{{ button_url }}"
-                                                                           data-button-weight="{{ button_details.weight }}"
-                                                                           class="btn btn-xs btn-{{ button_details.button_class }}"
-                                                                           title="{{ button_title }}">
-                                                                            <i class="mdi {{ button_details.icon_class }}"></i>
-                                                                        </a>
-                                                                    {% else %}
-                                                                        <a class="btn btn-xs btn-danger"
-                                                                           title="ERROR: Invalid link!">
-                                                                            <i class="mdi mdi-alert"></i>
-                                                                        </a>
+                                                        {% if item_details.buttons.items|length > 0 %}
+                                                            <div class="buttons pull-right">
+                                                                {% for button_title, button_details in item_details.buttons.items %}
```

Comment:
Probably should add code here to explicitly discard any instance of `NavMenuImportButton`, as apps that provide their own menu items may still have such buttons present.

Also, note that app menus *may* contain custom buttons, for example Circuit Maintenance:
![image](https://github.com/nautobot/nautobot/assets/5603551/49e3140e-d464-4a63-976b-833d81837c1d)

Does this resiliently handle such cases or does it break if there are menu items that contain more than one button?

Reply:
Thank you for the detailed answer!